jet/job-management/src/main/java/com/hazelcast/samples/jet/jobmanagement/JobSuspendResumeListener.java

```diff
@@ -33,12 +36,12 @@ public static void main(String[] args) throws InterruptedException {
         Job job = jet1.newJob(p, jobConfig);
 
         // Register JobStatusListener
-        job.addStatusListener(event ->
-                System.out.printf("Job status changed: %s -> %s. User requested? %b%n",
-                    event.getPreviousStatus(),
-                    event.getNewStatus(),
-                    event.isUserRequested())
-        );
+        job.addStatusListener(event -> {
+            System.out.printf("Job status changed: %s -> %s. User requested? %b%n",
+                    event.getPreviousStatus(), event.getNewStatus(), event.isUserRequested());
+
+            if (event.getNewStatus() == JobStatus.RUNNING) restartCount++;
```

Comment:
The initial `RUNNING` status might be missed and _restart count_ is not well-defined. What if I repeatedly restart a job before it even reaches the `RUNNING` status? In this implementation, the restart count would be zero although the job is not really running or pending (`NOT_RUNNING`).
1. Rename `restartCount` to `runCount` and initialize it to 1.
2. Add a separate status listener to increment run count. This listener should be added after ensuring that the job is running, i.e. `waitForStatus(job, JobStatus.RUNNING);`
3. Replace conditions `restartCount == n` with `runCount == n + 1`.

Reply:
I've implemented the suggested changes. Please keep in mind that the goal of the code samples is to provide an easily consumable and short demo of a specific feature for users. The examples are not meant to be complete solutions, and things like handling edge-cases can be considered out of scope.